utils/decision.py

```diff
@@ -0,0 +1,60 @@
+from enum import Enum, auto
+
+from utils.result import ResultTypes
+
+
+class DecisionStrategies(Enum):
+    Strict = auto()
+    Majority = auto()
+    Unanimous = auto()
+
+
+def decide(strategy: DecisionStrategies, results: list[ResultTypes]) -> ResultTypes:
+    """
+    Given a decision strategy and a list of results, it computes the decision.
+    """
+
+    match strategy:
+        case DecisionStrategies.Strict:
```

Comment:
is there a thing in python where you can attach a value (like a function) to an enum const? then we can add the _strict function etc to it and use it directly

Reply:
what if we switch to an abc and subclasses system instead? that way you also dont need the extra `decide` function cause it'd just be `strategy.decide(results)`